freqtrade/persistence/models.py

```diff
@@ -568,6 +569,37 @@ def calc_profit_ratio(self, rate: Optional[float] = None,
         profit_ratio = (close_trade_value / self.open_trade_value) - 1
         return float(f"{profit_ratio:.8f}")
 
+    def recalc_trade_from_orders(self):
+        # We need at least 2 orders for averaging amounts and rates.
+        if len(self.orders) < 2:
+            # Just in case, still recalc open trade value
+            self.recalc_open_trade_value()
+            return
+
+        total_amount = 0.0
+        total_stake = 0.0
+        for temp_order in self.orders:
+            if temp_order.ft_is_open or \
+                    temp_order.ft_order_side != 'buy' or \
+                    temp_order.status not in NON_OPEN_EXCHANGE_STATES:
```

Comment:
Personally, i like wrapping lines in paranteses better than splitting with `\` (which you'll observe, we're doing almost nowhere).